frontend/lib/resolution/Resolver.cpp

```diff
@@ -983,6 +983,50 @@ static bool isCallToPtr(const AstNode* formalTypeExpr) {
   return false;
 }
 
+// helper to gather bad actuals and report NoMatchingCandidates error
+static void
+handleRejectedCandidates(Context* context,
+                         ResolutionResultByPostorderID& byPostorder,
+                         std::vector<ApplicabilityResult>& rejected,
+                         const resolution::CallInfo& ci,
+                         const uast::Call*& call,
+                         const std::vector<const uast::AstNode*>& actualAsts) {
+  // By performing some processing in the resolver, we can issue a nicer error
+  // explaining why each candidate was rejected.
+  std::vector<const uast::VarLikeDecl*> actualDecls;
+  actualDecls.resize(rejected.size());
+  // check each rejected candidate for uninitialized actuals
+  for (size_t i = 0; i < rejected.size(); i++) {
+    auto &candidate = rejected[i];
+    if (/* skip computing the formal-actual map because it will go poorly
+           with an unknown formal. */
+        candidate.failedDueToWrongActual()) {
+      auto fn = candidate.initialForErr();
+      resolution::FormalActualMap fa(fn, ci);
+      auto& badPass = fa.byFormalIdx(candidate.formalIdx());
+      const uast::AstNode *actualExpr = nullptr;
+      const uast::VarLikeDecl *actualDecl = nullptr;
+      if (call && 0 <= badPass.actualIdx() &&
+          (size_t)badPass.actualIdx() < actualAsts.size()) {
+        actualExpr = actualAsts[badPass.actualIdx()];
+      }
+      // look for a definition point of the actual for error reporting of
+      // uninitialized vars typically in the case of bad split-initialization
+      if (actualExpr && actualExpr->isIdentifier()) {
+        auto &resolvedExpr = byPostorder.byAst(actualExpr->toIdentifier());
```

Comment:
`byAst` doesn't need you to cast `toIdentifier()` since the pointer return from `toWhatever` is the same (it's just a narrower type).